cilium/CFP-22215-mutual-auth-for-service-mesh.md

```diff
@@ -0,0 +1,442 @@
+# CFP-22215: Mutual Authentication for Cilium Service Mesh
+
+**SIG: SIG-ServiceMesh**
+
+**Begin Design Discussion:** 2022-10-28
+
+**Cilium Release:** 1.14
+
+**Authors:** Joe Stringer <joe@isovalent.com>, Nick Young <nick@isovalent.com>
+
+## Summary
+
+Enable Cilium to authenticate endpoints on peer nodes in the cluster and control dataplane connectivity based on successful mutual authentication.
+
+## Motivation
+
+Many users of existing service meshes consider pod mutual authentication to be a critical feature of a service mesh, and are looking for Cilium to provide an implementation of this feature as part of the Cilium Service Mesh offering.
+
+Existing mutual authentication implementations put several restrictions upon users in order to attain additional security guarantees. These can range from requiring apps to use TCP+TLS for authentication/encryption to requiring (sidecar) proxies in the dataplane. These restrictions increase the baseline complexity and processing cost of implementing mutual authentication. We believe that by combining the best of session-based authentication and network-based authentication, we can provide a secure implementation that is faster and more flexible than prior implementations.
+
+## Goals
+
+* Provide a mechanism for pods to opt into peer authentication
+* Pluggable for existing certificate management systems (SPIFFE, Vault, SMI, Istio, cert-manager, …)
+* Configurable certificate granularity
+* Leverage existing encryption protocol support for the dataplane
+
+
+## Non-Goals
+
+* Inline authentication via a proxy
+* Drop less connection establishment
```

Comment:
The space between drop and less confused me, did you mean something like drop-less?

Reply:
Yes, I've updated to make clearer, and also that we are going to _try_ to minimize the number of drops, even in the first cut.